runc.go

```diff
@@ -340,14 +340,14 @@ func (o *KillOpts) args() (out []string) {
 }
 
 // Kill sends the specified signal to the container
-func (r *Runc) Kill(context context.Context, id string, sig int, opts *KillOpts) error {
+func (r *Runc) Kill(context context.Context, id string, sig string, opts *KillOpts) error {
```

Comment:
I see the `go-runc` module is already at v1.0.0, and I see there's various projects using this module (outside of containerd); changing this signature would require incrementing the major version to v2.0.0.

Perhaps instead, we should take the same approach here as discussed in the containerd ticket;

- add a `Signal` (string) field to `KillOpts`
- if both `sig` and `KillOpts.Signal` are set, then the latter overrides the former
- callers can decide to either use `sig`, or `KillOpts.Signal`

Probably warrants an update to the GoDoc of this function to describe how it should be used (and that `KillOpts.Signal` should be considered in favor of `sig` by callers that do not want to (or can't) perform the conversion of string to signal (numeric), such as in the LCOW case.

Reply:
I agree with @thaJeztah. As like containerd, we should keep go-runc backward-compatible as much as we can.